PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/ruleConfigs.json

```diff
@@ -10,5 +10,11 @@
   "ServiceBusReceiverAsyncClientCheck": {
     "client_name": "ServiceBusReceiverAsyncClient",
     "antipattern_message": "Use of ServiceBusReceiverAsyncClient detected. Use ServiceBusProcessorClient instead."
+  },
+
+  "HardcodedAPIKeysAndTokensCheck": {
+    "clients_to_check": ["AzureKeyCredential", "AccessToken", "KeyCredential", "AzureNamedKeyCredential", "AzureSasCredential", "AzureNamedKey", "ClientSecretCredentialBuilder", "UsernamePasswordCredentialBuilder", "BasicAuthenticationCredential"],
```

Comment:
Why are we calling these clients_to_check, aren't these just different ways to authenticate?
we should consider renaming this to something more accurate.

Reply:
changed to services_to_check -- I also made a new getter for it. i thought about combining with the clients to check getter but I think its better to keep everything separate for now incase there's a situation where I have specific clients to look for in certain services (eg the single operations in loop check in the future will be expanded to storage & text analytics)